idaes/surrogate/keras_surrogate.py

```diff
@@ -16,7 +16,13 @@
 import numpy as np
 import json
 import os.path
-import tensorflow.keras as keras
+
+try:
+    import tensorflow.keras as keras
+    keras_available = True
+except:
+    keras_available = False

```

Comment:
You aren't testing that keras is available before using it, which means that a user that does not have tensorflow installed will just get a `NameError: name `keras` is not defined`.  I would recommend either adding a test for keras_available (and an appropriate exception), or just use Pyomo's `attempt_import()` which will generate the exception automatically when `keras` is used:
```
from pyomo.common.dependencies import attempt_import
keras, keras_available = attempt_import('tensorflow.keras`)
```

Reply:
I made this change.